code/game/objects/items/stacks/stack.dm

```diff
@@ -250,64 +282,149 @@
+	if(!istype(picked_recipe))
+		return
+
+	make_item(builder, picked_recipe, 1)
+
+/// Used as a callback for radial building.
+/obj/item/stack/proc/radial_check(mob/builder)
+	if(QDELETED(builder) || QDELETED(src))
+		return FALSE
+	if(builder.incapacitated())
+		return FALSE
+	if(!builder.is_holding(src))
+		return FALSE
+	return TRUE
+
+#undef FULL_LIST
+
+/// Makes the item with the given recipe.
+/obj/item/stack/proc/make_item(mob/builder, datum/stack_recipe/recipe, multiplier)
+	if(get_amount() < 1 && !is_cyborg) //sanity check as this shouldn't happen
+		qdel(src)
+		return
+	if(!is_valid_recipe(recipe, recipes)) //href exploit protection
+		return
+	if(!multiplier || (multiplier <= 0)) //href exploit protection
+		return
+	if(!building_checks(builder, recipe, multiplier))
+		return
+	if(recipe.time)
+		var/adjusted_time = 0
+		builder.balloon_alert(builder, "building...")
+		builder.visible_message(
+			span_notice("[builder] starts building \a [recipe.title]."),
+			span_notice("You start building \a [recipe.title]..."),
+		)
+		if(HAS_TRAIT(builder, recipe.trait_booster))
+			adjusted_time = (recipe.time * recipe.trait_modifier)
+		else
+			adjusted_time = recipe.time
+		if(!do_after(builder, adjusted_time, target = builder))
+			builder.balloon_alert(builder, "interrupted!")
+			return
+		if(!building_checks(builder, recipe, multiplier))
+			return
+
+	var/atom/created
+	if(recipe.max_res_amount > 1) // Is it a stack?
+		created = new recipe.result_type(builder.drop_location(), recipe.res_amount * multiplier)
+		builder.balloon_alert(builder, "built items")
+
+	else if(ispath(recipe.result_type, /turf))
+		var/turf/covered_turf = builder.drop_location()
+		if(!isturf(covered_turf))
+			return
+		covered_turf.PlaceOnTop(recipe.result_type, flags = CHANGETURF_INHERIT_AIR)
+		builder.balloon_alert(builder, "placed [ispath(recipe.result_type, /turf/open) ? "floor" : "wall"]")
+
+	else
+		created = new recipe.result_type(builder.drop_location())
+		builder.balloon_alert(builder, "built item")
+
+	if(created)
+		created.setDir(builder.dir)
+
+	// Use up the material
+	use(recipe.req_amount * multiplier)
+	builder.investigate_log("[key_name(builder)] crafted [recipe.title]", INVESTIGATE_CRAFTING)
+
+	// Apply mat datums
+	if(recipe.applies_mats && LAZYLEN(mats_per_unit))
+		if(isstack(created))
+			var/obj/item/stack/crafted_stack = created
+			crafted_stack.set_mats_per_unit(mats_per_unit, recipe.req_amount / recipe.res_amount)
+		else
+			created.set_custom_materials(mats_per_unit, recipe.req_amount / recipe.res_amount)
+
+	// We could be qdeleted - like if it's a stack and has already been merged
+	if(QDELETED(created))
+		return TRUE
+
+	// Add fingerprints first, otherwise created might already be deleted because of stack merging
+	created.add_fingerprint(builder)
+	if(isitem(created))
+		builder.put_in_hands(created)
+
+	//BubbleWrap - so newly formed boxes are empty
+	if(istype(created, /obj/item/storage))
```

Comment:
not related to your PR but this is a very weird way of handling this.

Reply:
Yeah I saw this code block when I moved it over and thought that was jank as hell. 
Future refactor though.